src/pages/productDetail/ProductDetail.page.jsx

```diff
@@ -0,0 +1,214 @@
+/* eslint-disable react/prop-types */
+import React, { useState } from "react";
+import { useNavigate, useParams } from "react-router-dom";
+import Button from "../../components/button/Button.component";
+import GalleryImages from "../../components/galleryImages/GalleryImages.component";
+import Loading from "../../components/loading/Loading.component";
+import Modal from "../../components/modal/Modal.component";
+import Tabs from "../../components/tabs/Tabs.component";
+import Tag from "../../components/tag/Tag.component";
+import { useDetailProduct } from "../../utils/hooks/useDetailProduct";
+import { storageCart } from "../../utils/storage";
+import {
+  StyledContainerDetail,
+  StyledImageCart,
+  StyledLabelInput,
+  StyledLabelName,
+  StyledLabelPrice,
+  StyledLabelReturn,
+  StyledLabelSku,
+  StyledWrapperCard,
+  StyledWrapperLabel,
+  StyledWrapperLeft,
+  StyledWrapperPrice,
+  StyledWrapperPriceDetail,
+  StyledWrapperPriceQuantity,
+  StyledWrapperQuantity,
+  StyledWrapperQuantityAll,
+  StyledWrapperRight,
+  StyledLabelReturnMobile,
+  StyledLabelNameDetails,
+  StyledLabelSkuDetails,
+  StyledLabelCategoryDetails,
+  StyledLabelPriceDetails,
+  StyledWrapperPriceDetails,
+  StyledWrapperTags,
+  StyledLabelAdded,
+} from "./ProductDetail.styles";
+
+function ProductDetailPage() {
+  const navigate = useNavigate();
+  const { productId } = useParams();
+  const data = useDetailProduct(productId);
+  const [stock, setStock] = useState(1);
+  const [isShowModal, setIsShowModal] = useState(false);
+  const productDetail = data.isLoading === false ? data.data.results[0] : [];
+  const productCart =
+    data.isLoading === false ? { ...productDetail, stockOnCart: stock } : [];
+  const isDisabledRemove = stock === 1 ? true : false;
+  const isDisabledAdd =
+    data.isLoading === false
+      ? stock === productDetail.data.stock
+        ? true
+        : false
+      : 1;
+
+  const isDisabledCart =
+    data.isLoading === false
+      ? productDetail.data.stock === 0
+        ? true
+        : false
+      : 1;
+
+  const handleReturn = () => {
+    navigate("/products");
+  };
+
+  const addItem = () => {
+    setStock(stock + 1);
+  };
+
+  const removeItem = () => {
+    setStock(stock - 1);
+  };
+
+  const handleAddToCart = () => {
+    setIsShowModal(true);
```

Comment:
We can add a context/redux and let it work along the localStorage in order to fulfil this criteria. This could also help to easier update information for components like the current available stock (originalStock - amountInCart)
We can take into account if this product already is in the cart, that way we prevent exceeding the StockAvailable when adding to the cart.